refactor ES_VERSIONS to keep itself up to date

0xdevalias opened the pull request:
- fixes https://github.com/g-plane/node-green/issues/2

0xdevalias committed:
refactor ES_VERSIONS to keep itself up to date

0xdevalias committed:
add lint:fix script to package.json

0xdevalias committed:
add tests for modern ES_VERSIONS

g-plane commented on package.json:
```diff
@@ -22,6 +22,7 @@
   "scripts": {
     "test": "jest",
     "lint": "eslint . -f=beauty --ext=.ts",
+    "lint:fix": "eslint . -f=beauty --ext=.ts --fix",
```
```suggestion
```

Avoid doing unrelated things.

g-plane commented on tests/index.test.ts:
```diff
@@ -103,3 +104,162 @@ test('with harmony', async () => {
+    })
+  })
+
+  test('ES2022', async () => {
```
There's no need to add every years.

0xdevalias replied:
I didn't add every year. I added 2 cases.

g-plane replied:
I think one year is enough.

0xdevalias replied:
You're welcome to push changes directly to the branch if you want to reduce the scope of the tests.

g-plane replied:
I'm busy, and you're working on this pull request.

0xdevalias replied:
I'm busy too.. The PR is perfectly valid with 2 years in the test cases. You didn't like that.. so you want me to put in extra work to align your stylistic choices. You could even do the whole thing through the UI and make a suggested change to remove it like you did before and then it would be a single button click for either of us to resolve, but you don't..?

0xdevalias replied:
Fixed with:

- `add missing test fixture` [`6b433db` (#3)](https://github.com/g-plane/node-green/pull/3/commits/6b433dba47d54432e78abe7f0e52092d841ef9ac)
- `reduce test reliability to satisfy code review` [`c37a10f` (#3)](https://github.com/g-plane/node-green/pull/3/commits/c37a10f051ca6f47e56fe26c144ce5d81374e41b)

0xdevalias committed:
Update package.json

Co-authored-by: Pig Fang <g-plane@hotmail.com>

0xdevalias committed:
add missing test fixture

0xdevalias committed:
reduce test reliability to satisfy code review

as forced during code review https://github.com/g-plane/node-green/pull/3/files#r1053923080
